static/app/types/auth.tsx

```diff
@@ -32,6 +32,10 @@ export type Authenticator = {
    */
   description: string;
   devices: AuthenticatorDevice[];
+  /**
+   * New enrollments of this 2FA interface are not allowed
+   */
+  disallowNewEnrollment: boolean;
```

Comment:
A test case in would be nice `https://github.com/getsentry/sentry/blob/master/tests/js/spec/views/accountSecurity.spec.jsx` :D